File: 600/686 - Repeated String Match.java
class Solution {
    public int repeatedStringMatch(String A, String B) {
        String r = A;
        int max = B.length() / A.length() + 2;
        
        for (int i = 1; i <= max; ++i) {
            if (r.contains(B)) {
                return i;
            }
            
            r += A;
        }
        
        return -1;
    }
}

File: csharp/686.repeated-string-match.cs
/*
 * @lc app=leetcode id=686 lang=csharp
 *
 * [686] Repeated String Match
 */

// @lc code=start
public class Solution {
    public int RepeatedStringMatch(string A, string B) {
        int m = A.Length, n = B.Length;
        for (int i = 0; i < m; ++i) {
            int j;
            for (j = 0; j < n && B[j] == A[(i + j) % m]; ++j);
            if (j == n) {
                return 1 + (i + j - 1) / m;
            }
        }
        return -1;
    }
}
// @lc code=end

